### PR TITLE
OBSDOCS-953/TRACING-4062 Write documentation and technical enablement for journald component

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is tied to a product release date, which is currently set to **June 5** (Wednesday).

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Edit of https://github.com/openshift/openshift-docs/pull/74207

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

This PR should be published as part of RHOSDT 3.2 release

Issue:

https://issues.redhat.com/browse/OBSDOCS-953
https://issues.redhat.com/browse/TRACING-4062

Link to docs preview:

https://75257--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#journald-receiver_otel-configuration-of-otel-collector

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -14,7 +14,7 @@ Receivers get data into the Collector.
 [id="otlp-receiver_{context}"]
 === OTLP Receiver
 
-The OTLP receiver ingests traces and metrics using the OpenTelemetry protocol (OTLP).
+The OTLP receiver ingests traces and metrics by using the OpenTelemetry Protocol (OTLP).
```
Comment (pavolloffay) [inline modules/otel-collector-components.adoc:17]: It can ingest logs as well

**Review by IshwarKanse — APPROVED** [review]

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -199,6 +199,119 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
 Wildcards with `+*+` are accepted under the `cors_allowed_origins`.
 To match any origin, enter only `+*+`.
 
+[id="journald-receiver_{context}"]
+=== Journald receiver
+
+The Journald receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```
Comment (mburke5678) [inline modules/otel-collector-components.adoc:205]: We just lower-case journald in the docs. 

```suggestion
The journald receiver is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```

File: modules/otel-collector-components.adoc [hunk 3]
```diff
@@ -199,6 +199,119 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+  tolerations:
+  - key: node-role.kubernetes.io/master
+    operator: Exists
+    effect: NoSchedule
+EOF
+----
+<1> Filters output by message priorities or priority ranges. Defaults to `info`.
+<2> Lists the units to read entries from. If empty, entries are read from all units.
+<3> Includes very long logs and logs with unprintable characters. Defaults to `false`.
+<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encounters an error from downstream components. Defaults to `false`.
```
Comment (mburke5678) [inline modules/otel-collector-components.adoc:310]: I think you are missing a word here. 
```suggestion
<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when <MISSING WORD?>encounters an error from downstream components. Defaults to `false`.
```

File: modules/otel-collector-components.adoc (hunk 2 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:205]: Thank you. I'm actually capitalizing all the component names now, so no need to worry about this one.

Comment (max-cx) [reply modules/otel-collector-components.adoc:205]: However, there is one more instance that I am lowercasing. Thank you again.

File: modules/otel-collector-components.adoc (hunk 3 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:311]: :slightly_smiling_face: thank you. I think it's meant to be `encountering`, not `encounters`. Fixed.

**Comment by max-cx** [comment]
Closing this PR because I cherry-picked its commit into https://github.com/openshift/openshift-docs/pull/76340.

File: modules/otel-collector-components.adoc [hunk 4]
```diff
@@ -199,6 +199,120 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+  tolerations:
+  - key: node-role.kubernetes.io/master
+    operator: Exists
+    effect: NoSchedule
+EOF
+----
+<1> Filters output by message priorities or priority ranges. Defaults to `info`.
+<2> Lists the units to read entries from. If empty, entries are read from all units.
+<3> Includes very long logs and logs with unprintable characters. Defaults to `false`.
+<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encountering an error from downstream components. Defaults to `false`.
```
Comment (skrthomas) [inline modules/otel-collector-components.adoc:311]:
```suggestion
<1> The priorities or priority ranges to filters output by . The defaults is `info`.
<2> The list of units to read entries from. If empty, entries are read from all units.
<3> If the specified value is `true`, very long logs and logs with unprintable characters are included. The default is `false`.
<4> If set to `true`, the receiver pauses reading a file and attempts to resend the current batch of logs when encountering an error from downstream components. The default is `false`.
```

File: modules/otel-collector-components.adoc [hunk 5]
```diff
@@ -199,6 +199,120 @@ The OpenCensus receiver provides backwards compatibility with the OpenCensus pro
...
+<5> The time interval to wait after the first failure before retrying. Defaults to `1s`. The units are `ms`, `s`, `m`, `h`.
+<6> The upper bound for the retry backoff interval. When this value is reached, the time interval between consecutive retry attempts remains constant at this value. Defaults to `30s`. The supported units are `ms`, `s`, `m`, `h`.
+<7> The maximum time interval, including retry attempts, for attempting to send a logs batch to a downstream consumer. When this value is reached, the data are discarded. If the set value is `0`, retrying never stops. Defaults to `5m`. The supported units are `ms`, `s`, `m`, `h`.
```
Comment (skrthomas) [inline modules/otel-collector-components.adoc:314]:
```suggestion
<7> The maximum time interval, including retry attempts, for attempting to send a logs batch to a downstream consumer. When this value is reached, the data are discarded. If the set value is `0`, retrying never stops. The defaults is `5m`. The supported units are `ms`, `s`, `m`, `h`.
```

**Review by skrthomas — COMMENTED** [review]
A few code callouts that need to be converted to complete sentences, otherwise lgtm and is OK to merge on 5/29

File: modules/otel-collector-components.adoc (hunk 4 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:311]: Thank you. I applied some of these suggestions.

File: modules/otel-collector-components.adoc (hunk 5 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc]: Thank you. Applied.

**Pull request merged**